src/ess/reduce/scripts/grow_nexus.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+from typing import Optional
+
+import h5py
+
+
+def _scale_group(event_data: h5py.Group, scale: int):
+    event_index = (event_data['event_index'][:] * scale).astype('uint')
+    event_data['event_index'][:] = event_index
+
+    size = event_data['event_id'].size
+    event_data['event_id'].resize(event_index[-1], axis=0)
+    event_data['event_time_offset'].resize(event_index[-1], axis=0)
+
+    for s in range(1, scale):
+        event_data['event_id'][s * size : (s + 1) * size] = event_data['event_id'][
+            :size
+        ]
+        event_data['event_time_offset'][s * size : (s + 1) * size] = event_data[
+            'event_time_offset'
+        ][:size]
+
+
+def _grow_nexus_file_impl(file: h5py.File, detector_scale: int, monitor_scale: int):
+    for group in file.values():
+        if group.attrs.get('NX_class', '') == 'NXentry':
+            entry = group
+            break
+    for group in entry.values():
+        if group.attrs.get('NX_class', '') == 'NXinstrument':
+            instrument = group
+            break
+    for group in instrument.values():
+        if (nx_class := group.attrs.get('NX_class', '')) in (
+            'NXdetector',
+            'NXmonitor',
+        ):
+            for subgroup in group.values():
+                if subgroup.attrs.get('NX_class', '') == 'NXevent_data':
+                    _scale_group(
+                        subgroup,
+                        scale=detector_scale
+                        if nx_class == 'NXdetector'
+                        else monitor_scale,
+                    )
+
+
+def grow_nexus_file(
+    *, filename: str, detector_scale: int, monitor_scale: Optional[int]
+):
+    with h5py.File(filename, 'a') as f:
+        _grow_nexus_file_impl(
+            f,
+            detector_scale,
+            monitor_scale if monitor_scale is not None else detector_scale,
+        )
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
Sorry I've only noticed this now, but I would suggest to maybe return a new file, instead of modifying the input file in-place?
This would mean we also need an output file argument.

I don't know if this would make usability worse?
`grow_nexus -f small.nxs -o large.nxs -s 2`
vs
`grow_nexus myfile.nxs 2`